### ISSUE TITLE
How to get browser requests working?

### ISSUE THREAD

**Issue by cocowalla** [AzureAD/azure-activedirectory-library-for-cordova#163]
We're using ADAL in a Cordova app, and it's working great for when we make web service calls in our code. We get the token, then we put it in the `Authorization` header - no problems there.

However, requests that the browser (e.g. Android WebView) makes, for example for images (using the HTML `img` tag) or audio (using the HTML5 audio tag), do not work.

Using the Chrome debugger, I see 2 HTTP requests being made:

1. The call to the specified URL. The server (hosted on Azure) responds with a 302 pointing to a rather long `https://login.microsoftonline.com/MY_TENANT_ID/oauth2/authorize/........` oauth2 URL
2. A call is made to the URL that was returned in the 302, to which the server responds with an HTML response

Is this supposed to work?

**Comment by matrosov-nikita** [comment]
@cocowalla, thanks for reporting, I'll take a look at this a little later.

**Comment by matrosov-nikita** [comment]
@cocowalla, sorry for the delay, just to clarify these HTTP requests
1) The first one (`/authorize` endpoint) is used to request an authorization code
2) The second (`/token` endpoint) is used to request access token (headers include authorization code that you acquired in the previous response)

As for Webview requests, can you share any sample HTML which we can refer to understand the problem?

**Comment by cocowalla** [comment]
@matrosov-nikita Yes, I'm clear on what the endpoints do.

Regarding HTML, just about anything will do. For example:

`<img src="https://my-oauth-projected-site/img/something.jpg">`

I need to access images and audio files that are stored on a server which is protected by OAuth. Now, I know I can access these fine in code, as I have control over the headers which are sent (so I can sent the Authorization header).

But to keep things simple, I'd _much_ prefer in this case to simply point to the protected images and audio files in HTML (like the `img` tag above). The problem is that I have no control over the headers sent by the Webview. I had expected that the ADAL Cordova plugin would have some mechanism for injecting a token into an Authorization header which would be sent by the Webview, but I can't see anything?

**Comment by matrosov-nikita** [comment]
@cocowalla Yes, ADAL plugin doesn't provide this opportunity. My guess is that it's pretty difficult to determine when we need to include custom authentication header with access token in the Webview request. This would allow for the possibility of sending a token to the third-party server.

**Comment by cocowalla** [comment]
@matrosov-nikita Firstly, you should be able to toggle this feature on/off, as it's not going to be applicable for all use cases.

As for *when* to send the header, it shouldn't be too difficult. A couple of ways would be:

1. From your app code, provide the plugin with a list of domains or base URLs for which you want an Authorization header to be sent
2. From your app code, provide the plugin with a list of regex expressions for which you want matching URLs to have an Authorization header sent

**Comment by matrosov-nikita** [comment]
@cocowalla, thanks, the first approach looks good to me, I'll discuss it and let you know once we decide to implement it.
